docs/guides/go.md

```diff
@@ -311,112 +310,89 @@ information on why the transaction was rejected.
+            txs[i] = &abcitypes.ExecTxResult{}
+            if _, err := txs[i].MarshalToSizedBuffer(nil); err != nil {
+                log.Panicf("Error marshalling tx result: %v", err)
+            }
+        }
+    }
+
+    return &abcitypes.ResponseFinalizeBlock{
+        TxResults:        txs,
+        AppHash:          []byte{},
+        Events:           []abcitypes.Event{},
+        ValidatorUpdates: abcitypes.ValidatorUpdates{},
+    }, nil
 }
 ```
 
-Next, let's modify `DeliverTx` to add the `key` and `value` to the database transaction every time our application
-receives a new application transaction through `RequestDeliverTx`.
-
-```go
-func (app *KVStoreApplication) DeliverTx(req abcitypes.RequestDeliverTx) abcitypes.ResponseDeliverTx {
-	if code := app.isValid(req.Tx); code != 0 {
-		return abcitypes.ResponseDeliverTx{Code: code}
-	}
-
-	parts := bytes.SplitN(req.Tx, []byte("="), 2)
-	key, value := parts[0], parts[1]
-
-	if err := app.onGoingBlock.Set(key, value); err != nil {
-		log.Panicf("Error writing to database, unable to execute tx: %v", err)
-	}
-
-	return abcitypes.ResponseDeliverTx{Code: 0}
-}
-```
-
-Note that we check the validity of the transaction _again_ during `DeliverTx`.
-Transactions are not guaranteed to be valid when they are delivered to an
-application, even if they were valid when they were proposed.
-This can happen if the application state is used to determine transaction
-validity. Application state may have changed between the initial execution of `CheckTx`
-and the transaction delivery in `DeliverTx` in a way that rendered the transaction
-no longer valid.
-
-`EndBlock` is called to inform the application that the full block has been delivered
-and give the application a chance to perform any other computation needed, before the
-effects of the transactions become permanent.
-
-Note that `EndBlock` **cannot** yet commit the Badger transaction we were building
-in during `DeliverTx`.
-Since other methods, such as `Query`, rely on a consistent view of the application's
-state, the application should only update its state by committing the Badger transactions
-when the full block has been delivered and the `Commit` method is invoked.
-
 The `Commit` method tells the application to make permanent the effects of
 the application transactions.
 Let's update the method to terminate the pending Badger transaction and
 persist the resulting state:
 
 ```go
-func (app *KVStoreApplication) Commit() abcitypes.ResponseCommit {
-	if err := app.onGoingBlock.Commit(); err != nil {
-		log.Panicf("Error writing to database, unable to commit block: %v", err)
-	}
-	return abcitypes.ResponseCommit{Data: []byte{}}
+func (app KVStoreApplication) Commit(_ context.Context, commit *abcitypes.RequestCommit) (*abcitypes.ResponseCommit, error) {
+    if err := app.onGoingBlock.Commit(); err != nil {
+        return &abcitypes.ResponseCommit{}, err
+    }
+    return &abcitypes.ResponseCommit{}, nil
```

Comment:
```suggestion
    return &abcitypes.ResponseCommit{}, app.onGoingBlock.Commit()
```

Same comment as in the `go-built-in` version: the non-`nill` err will cause ComtBFT to panic, but I find it easier to follow if the panic appears here explicitly (the reader does not necessarily know that CometBFT panics with non-`nil` error returned by ABCI calls).